### PR TITLE
remove OptionsExternalVersion

### LINKED ISSUE #19434: Remove optionsExternalVersion after #17922 gets merged
Just a reminder to remove the annoying `optionsExternalVersion` after #17922 lands.

### PR DESCRIPTION
**Pull request by alexzielenski**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR simply respects an old [TODO comment](https://github.com/kubernetes/kubernetes/blob/10c37c5942399e55290c88606243e638d92008a4/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go#L71)'s wishes and removes `OptionsExternalVersion`. Instead of allowing this to be configurable, the builtin group version is used instead unconditionally (previously the default).

Years ago it seemed the intention was to remove this: https://github.com/smarterclayton/kubernetes/commit/96a1674da2399623a53207421851d744d40f6aa2#r15308641

but unfortunately a dependency from `ThirdPartyResourceVersion` [prevented removal](https://github.com/kubernetes/kubernetes/issues/19434#issuecomment-174740531). 

Now that I'm not seeing that dependency anywhere I figure it would be nice to remove unnecessary code.

#### Which issue(s) this PR fixes:

Fixes #19434

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go [hunk 1]
```diff
@@ -280,6 +275,7 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		versionedList = indirectArbitraryPointer(versionedListPtr)
 	}
 
+	optionsExternalVersion := &schema.GroupVersion{Version: "v1"}
```
[the remark this answers is not in the document]
Comment (Jefftree) [reply staging/src/k8s.io/apiserver/pkg/endpoints/installer.go:278]: should we change this variable name to represent the "v1" more accurately?

**Comment by alexzielenski** [comment]
Comments addressed.

**Comment by alexzielenski** [comment]
/assign @smarterclayton 

I think you had some involvement with this field's original authorship.

**Pull request closed without merging**